src/sciline/_provider.py

```diff
@@ -220,7 +229,10 @@ def keys(self) -> Generator[Key, None, None]:
 
     def bind_type_vars(self, bound: dict[TypeVar, Key]) -> ArgSpec:
         """Bind concrete types to TypeVars."""
-        return self.map_keys(lambda arg: _bind_free_typevars(arg, bound=bound))
+        out = self.map_keys(lambda arg: _bind_free_typevars(arg, bound=bound))
+        if self._return is not None:
+            out._return = _bind_free_typevars(self._return, bound=bound)
+        return out
```

Comment:
Should this be moved into `map_keys`?

Reply:
See above. In Sciline, 'key' refers to the things that connect providers, i.e., domain types. So it applies to the return type as well.

But my question was in particular about whether it always makes sense to map the return key.